server/memphis_handlers_tags.go

```diff
@@ -0,0 +1,482 @@
+		return err
+	}
+	return nil
+}
+
+func AddTagToEntity(name string, to string, to_name string, background_color string, text_color string) error {
+	name = strings.ToLower(name)
+	exist, tag, err := IsTagExist(name)
+	if err != nil {
+		return err
+	}
+	if !exist {
+		err = CreateTag(name, to, to_name, background_color, text_color)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+	switch to {
+	case "station":
+		station_name, err := StationNameFromStr(to_name)
+		if err != nil {
+			return err
+		}
+		exist, station, err := IsStationExist(station_name)
+		if err != nil {
+			return err
+		}
+		if !exist {
+			return errors.New("Station does not exist")
+		}
+		_, err = tagsCollection.UpdateOne(context.TODO(), bson.M{"_id": tag.ID}, bson.M{"$addToSet": bson.M{"stations": station.ID}})
+		if err != nil {
+			return err
+		}
+
+		// case "schema":
+		// 	exist, schema, err := IsSchemaExist(to_name)
+		// 	if err != nil {
+		// 		return err
+		// 	}
+		// 	if !exist {
+		// 		return errors.New("Schema with this name does not exist")
+		// 	}
+		// 	_, err = tagsCollection.UpdateOne(context.TODO(), bson.M{"_id": tag.ID}, bson.M{"$addToSet": bson.M{"schemas": schema.ID}})
+		// 	if err != nil {
+		// 		return err
+		// 	}
+		// case "user":
+		// 	exist, user, err := IsUserExist(to_name)
+		// 	if err != nil {
+		// 		return err
+		// 	}
+		// 	if !exist {
+		// 		return errors.New("User with this name does not exist")
+		// 	}
+		// 	_, err = tagsCollection.UpdateOne(context.TODO(), bson.M{"_id": tag.ID}, bson.M{"$addToSet": bson.M{"users": user.ID}})
+		// 	if err != nil {
+		// 		return err
+		// 	}
+	}
+	return nil
+}
+
+func DeleteTagsByStation(name string) {
+	station_name, err := StationNameFromStr(name)
+	if err != nil {
+		serv.Errorf("Failed deleting tags: %v", err.Error())
+		return
+	}
+	exist, station, err := IsStationExist(station_name)
+	if err != nil {
+		serv.Errorf("Failed deleting tags: %v", err.Error())
+		return
+	}
+	if !exist {
+		serv.Warnf("Station does not exist")
+		return
+	}
+	_, err = tagsCollection.UpdateMany(context.TODO(), bson.M{}, bson.M{"$pull": bson.M{"stations": station.ID}})
+	if err != nil {
+		serv.Errorf("Failed deleting tags: %v", err.Error())
+		return
+	}
+}
+
+// func DeleteTagsBySchema(name string) {
+// 	exist, schema, err := IsSchemaExist(name)
+// 	if err != nil {
+// 		serv.Errorf("Failed deleting tags: %v", err.Error())
+// 		return
+// 	}
+// 	if !exist {
+// 		serv.Errorf("Schema with this name does not exist")
+// 		return
+// 	}
+// 	_, err = tagsCollection.UpdateMany(context.TODO(), bson.M{}, bson.M{"$pull": bson.M{"schemas": schema.ID}})
+// 	if err != nil {
+// 		serv.Errorf("Failed deleting tags: %v", err.Error())
+// 		return
+// 	}
+// }
+
+func DeleteTagsByUser(name string) {
+	exist, user, err := IsUserExist(name)
+	if err != nil {
+		serv.Errorf("Failed deleting tags: %v", err.Error())
+		return
+	}
+	if !exist {
+		serv.Errorf("User with this name does not exist")
```

Comment:
change to warn.
Let's clear something error is cases where unexpected things happened. mongo is unresponsive for example.
warn is for cases where there is an error managed by Memphis, for example duplicate entities

Reply:
please go over the entire code and fix where needed